### PR TITLE
Use mage to setup development environment

### LINKED ISSUE #1866: Use mage instead of make
#### Summary

Do not depend on `make` for setting up a development environment. @rvolosatovs Did some initial work with #1861, this is a follow-up.

#### What do you see now?

Mage targets instead, e.g.

```bash
$ ./mage dev:dbStart
```

and similar targets

#### Environment

all

#### Can you do this yourself and submit a Pull Request?

Yes

### PR DESCRIPTION
**Pull request by neoaggelos**
#### Summary

Related to #1866 

#### Changes

- Added mage targets `dev:dbStart`, `dev:dbStop`, `dev:dbErase`, `dev:dbSQL`, `dev:dbRedisCli` and dependencies
- Updated setup instructions in `DEVELOPMENT.md`

#### Notes for Reviewers

- Also fixed a few inconsistencies `mage` vs `./mage` in DEVELOPMENT.md
- Should we remove the makefile targets altogether?
- travis still uses `dev.databases.start`. We should update that too in `.travis.yml`

#### Checklist

- [X] Scope: The referenced issue is addressed, there are no unrelated changes.
- [X] Compatibility: The changes are backwards compatible with existing API, database and configuration, according to the stability commitments in `README.md`.
- [x] Testing: The changes are covered with unit tests. The changes are tested manually as well.
- [X] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [X] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: .mage/dev.go [hunk 1]
```diff
@@ -68,8 +69,72 @@ func (Dev) Misspell() error {
...
+	devDataDir            = ".dev/data"
+	devDatabaseName       = "ttn_lorawan_dev"
+	devDockerComposeFlags = []string{"-p", "lorawan-stack-dev"}
+)
+
+func dockerComposeFlags(args ...string) []string {
+	return append(devDockerComposeFlags, args...)
+}
+
+func execDC(args ...string) error {
```
Comment (htdvisser) [inline .mage/dev.go]: Probably better to call this `execDockerCompose` in full, since `execDC` isn't obvious in this package.

File: .mage/dev.go [hunk 2]
```diff
@@ -68,8 +69,72 @@ func (Dev) Misspell() error {
...
+
+// DBStart starts the databases of the development environment
+func (Dev) DBStart() error {
+	if mg.Verbose() {
+		fmt.Printf("Starting dev databases\n")
+	}
+	if err := execDC(append([]string{"up", "-d"}, devDatabases...)...); err != nil {
+		return err
+	}
+
```
Comment (htdvisser) [inline .mage/dev.go]: Nit: no need for spacing in small funcs

File: .mage/dev.go [hunk 3]
```diff
@@ -68,8 +69,72 @@ func (Dev) Misspell() error {
...
+func dockerComposeFlags(args ...string) []string {
+	return append(devDockerComposeFlags, args...)
+}
+
+func execDC(args ...string) error {
+	_, err := sh.Exec(nil, os.Stdout, os.Stderr, "docker-compose", dockerComposeFlags(args...)...)
+	return err
+}
+
+// DBStart starts the databases of the development environment
```
Comment (htdvisser) [inline .mage/dev.go]: Comments end with a period.
also in other places

File: .mage/dev.go [hunk 4]
```diff
@@ -68,8 +69,72 @@ func (Dev) Misspell() error {
...
+	mg.Deps(Dev.DBStart)
+	if mg.Verbose() {
+		fmt.Printf("Starting Redis-CLI shell\n")
+	}
+	return execDC("exec", "redis", "redis-cli")
+}
+
 // InitStack initializes the Stack.
 func (Dev) InitStack() error {
+	mg.Deps(Dev.DBStart)
```
Comment (htdvisser) [inline .mage/dev.go]: Not everyone uses the databases in docker, so let's not force this.

File: DEVELOPMENT.md [hunk 5]
```diff
@@ -40,25 +40,25 @@ This starts a CockroachDB and Redis database in Docker containers, creates a dat
...
 ```bash
-$ make dev.databases.start # Starts all databases in a Docker container
-$ make dev.databases.stop  # Stops all databases
+$ ./mage dev:dbStart # Starts all databases in a Docker container
+$ ./mage dev:dbStop  # Stops all databases
 
 # The contents of the databases will be saved in .dev/data.
 
-$ make dev.databases.erase # Stop all databases and erase storage.
+$ ./mage dev:dbErase # Stop all databases and erase storage.
```
Comment (htdvisser) [inline DEVELOPMENT.md]: Let's make the comment consistent with the ones above while we're making changes.

> Stops all databases and erases storage.

And add periods to the comments above.

File: .mage/dev.go (hunk 4 above)
Comment (neoaggelos) [reply .mage/dev.go]: Followed previous behaviour, can remove. Will need to update DEVELOPMENT.md instructions accordingly

**Comment by neoaggelos** [comment]
Updated based on comments. Removed old makefile targets.

There still is `make git.diff` that is being used, but that should be done in a separate PR, I think.

Please take a look at the `.travis.yml` change to make sure that we don't break anything.

**Pull request merged**